### PR TITLE
Sentry Alert configuration based on production and development environment

### PR DESCRIPTION
**Pull request by zeeshanrafiqrana**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by zeeshanrafiqrana** [comment]
Issue Associated with this PR: [Sentry Alert configuration based on production and development environment](https://github.com/matrix-org/synapse/issues/16732)

File: synapse/app/_base.py [hunk 1]
```diff
@@ -662,9 +662,14 @@ def setup_sentry(hs: "HomeServer") -> None:
 
     import sentry_sdk
 
+    # Set a default environment if it's not configured
+    default_environment = ""
+    environment = hs.config.metrics.sentry_environment or default_environment
```
Comment (erikjohnston) [inline synapse/app/_base.py:667]: I think the default for `environment` is `None`, so we probably should keep `None` as the default rather than `""`?

Comment (zeeshanrafiqrana) [reply synapse/app/_base.py:667]: @erikjohnston I need to test with None because I'm not sure right now. I tested with an empty string, and that condition worked. I will update you in a while after taking measures about the no environment setup.

Comment (zeeshanrafiqrana) [reply synapse/app/_base.py:667]: @erikjohnston Changes done and tested.
Please review again

**Comment by erikjohnston** [comment]
(Sign off has been completed out-of-band)

File: synapse/app/_base.py [hunk 2]
```diff
@@ -662,9 +662,14 @@ def setup_sentry(hs: "HomeServer") -> None:
...
     import sentry_sdk
 
+    # Set a default environment if it's not configured
+    default_environment = None
+    environment = hs.config.metrics.sentry_environment or default_environment
+
     sentry_sdk.init(
         dsn=hs.config.metrics.sentry_dsn,
         release=SYNAPSE_VERSION,
+        environment=environment,
```
Comment (erikjohnston) [inline synapse/app/_base.py:672]: I think this can now be simplified to:

```suggestion
    sentry_sdk.init(
        dsn=hs.config.metrics.sentry_dsn,
        release=SYNAPSE_VERSION,
        environment=hs.config.metrics.sentry_environment,
```

**Review by erikjohnston — CHANGES_REQUESTED** [review]
Thanks! Sorry, I've missed the fact that'll you need to:

1. add a changelog file (as per the [contributing guide](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog)); and
2. add a brief description of the new option at https://github.com/matrix-org/synapse/blob/10ada2ff6d2a08108edf5b4dbe6562cc9465523d/docs/usage/configuration/config_documentation.md#sentry

Hopefully that should only take ten minutes.

File: synapse/app/_base.py (hunk 2 above)
Comment (zeeshanrafiqrana) [reply synapse/app/_base.py]: @erikjohnston recommended changes are done. please review again. Thanks

Its my first open source contribution. Next I will remember all those steps

File: docs/usage/configuration/config_documentation.md [hunk 3]
```diff
@@ -2772,7 +2772,7 @@ enable_metrics: true
 ### `sentry`
 
 Use this option to enable sentry integration. Provide the DSN assigned to you by sentry
-with the `dsn` setting.
+with the `dsn` setting and specify the environment in the `environment` field. This allows for log maintenance based on different environments, ensuring better organization and analysis..
```
Comment (erikjohnston) [inline docs/usage/configuration/config_documentation.md:2775]:
```suggestion
with the `dsn` setting. 

 An optional `environment` field can be used to specify an environment. This allows
 for log maintenance based on different environments, ensuring better organization
 and analysis..
```

File: changelog.d/16738.feature [hunk 4]
```diff
@@ -0,0 +1 @@
+Enhanced Sentry configuration with a startup-loaded environment for improved system monitoring. Contributed by @zeeshanrafiqrana.
```
Comment (erikjohnston) [inline changelog.d/16738.feature:1]:
```suggestion
Add new Sentry configuration option `environment` for improved system monitoring. Contributed by @zeeshanrafiqrana.
```

**Review by erikjohnston — APPROVED** [review]
I've updated the wording a little bit to match our style, but they're only minor tweaks,

Thanks for your contribution!
